File: AppBackend/apps/core/enums.py

from django.db import models
from django.utils.translation import gettext_lazy as _

class GenderChoices(models.TextChoices):
    """Перечисление для пола"""
    MALE = 'Male', _('Мужской')  
    FEMALE = 'Female', _('Женский')  

class QuantityTypeChoices(models.TextChoices):
    """Перечисление для типов количества"""
    GRAMS = 'grams', _('Граммы')
    MILLILITERS = 'milliliters', _('Миллилитры')
    LITERS = 'liters', _('Литры')
    PIECES = 'pieces', _('Штуки')
    TABLESPOONS = 'tablespoons', _('Столовые ложки')
    TEASPOONS = 'teaspoons', _('Чайные ложки')
    CUPS = 'cups', _('Стаканы')

File: to_be_renaimed/lib/models/enums.dart

enum QuantityType {
  grams,
  milliliters,
  liters,
  pieces,
  tablespoons,
  teaspoons,
  cups;

  factory QuantityType.fromString(String value) {
    return QuantityType.values.firstWhere(
          (e) => e.toString().split('.').last == value.toLowerCase(),
      orElse: () => QuantityType.grams,
    );
  }

  String toDisplayString() {
    switch (this) {
      case QuantityType.grams:
        return 'Граммы (г)';
      case QuantityType.milliliters:
        return 'Миллилитры (мл)';
      case QuantityType.liters:
        return 'Литры (л)';
      case QuantityType.pieces:
        return 'Штуки (шт)';
      case QuantityType.tablespoons:
        return 'Столовые ложки (ст. л.)';
      case QuantityType.teaspoons:
        return 'Чайные ложки (ч. л.)';
      case QuantityType.cups:
        return 'Стаканы';
    }
  }

  String getShortName() {
    switch (this) {
      case QuantityType.grams:
        return 'г';
      case QuantityType.milliliters:
        return 'мл';
      case QuantityType.liters:
        return 'л';
      case QuantityType.pieces:
        return 'шт';
      case QuantityType.tablespoons:
        return 'ст. л.';
      case QuantityType.teaspoons:
        return 'ч. л.';
      case QuantityType.cups:
        return 'стак.';
    }
  }
}


enum Gender {
  male,
  female;

  factory Gender.fromString(String value) {
    return Gender.values.firstWhere(
          (e) => e.toString().split('.').last == value.toLowerCase(),
      orElse: () => Gender.male,
    );
  }

  String toDisplayString() {
    switch (this) {
      case Gender.male:
        return 'Мужской';
      case Gender.female:
        return 'Женский';
    }
  }

  String toPostgreSqlValue() {
    final value = toString().split('.').last;
    return value[0].toUpperCase() + value.substring(1).toLowerCase();
  }
}